src/sagas.js

```diff
@@ -241,6 +253,38 @@ function* fetchAllVms (action) {
   yield put(persistState())
 }
 
+function* fetchAllPools (action) {
+  const allPools = yield callExternalAction('getAllPools', Api.getAllPools, action)
+
+  if (allPools && allPools['vm_pool']) { // array
+    const internalPools = allPools.vm_pool.map(pool => Api.poolToInternal({ pool }))
+
+    const poolIdsToPreserve = internalPools.map(pool => pool.id)
+    yield put(removeMissingPools({ poolIdsToPreserve }))
+
+    yield put(updatePools({ pools: internalPools }))
+    yield put(updateVmsPoolsCount())
```

Comment:
The vmsCount updated here seems like a derived value - is it really needed?
Maybe there's a reason for it. But if not, I would like to avoid such a pre-processing since it increases complexity for future maintenance.

If we need it, can be the update triggered from UPDATE_POOLS reducer without the need for additional action?